CMakeLists.txt

```diff
@@ -28,20 +28,31 @@ if(NOT CMAKE_BUILD_TYPE)
 		"Debug" "Release" "MinSizeRel" "RelWithDebInfo")
 endif()
 
-set(CARGO_FLAGS "")
+
+if (NOT DEFINED CARGO_FLAGS)
```

Comment:
This makes CARGO_FLAGS always defined. So test `if(DEFINED CARGO_FLAGS)` also always passes and with empty `CARGO_FLAGS` build fails:
```
[100%] Running RUSTFLAGS= cargo  build --release;--features=logger-autoinit;--target=;--manifest-path=/home/runner/work/zenoh-c/zenoh-c/Cargo.toml
error: target was empty
Error: gmake[2]: *** [CMakeFiles/cargo.dir/build.make:71: CMakeFiles/cargo] Error 101
```

Reply:
Dear @milyin 

I set the flag set(CARGO_FLAGS "") as default value. 
I add  a variable called CUSTOM_TARGET used for specifing the cross-compilation  target.  Besides, CARGO_CHANNEL that  users can use for  compiling zenoh against different releases of rust. This is important for *( TIER-3 ) custom targets like in my case.  

Finally I add small section in the documentation to explain how to cross-compile for aarch64-unknown-linux-gnu. I hope it is sufficient for many use cases. 

Please let me know. I am willing to support in this regard. 

Best
Diego.